addons/sale/models/account_move_line.py

```diff
@@ -112,6 +115,13 @@ def _sale_create_reinvoice_sale_line(self):
                         # store it in the map_move_sale_line map
                         map_move_sale_line[move_line.id] = len(sale_line_values_to_create) - 1  # save the index of the value to create sale line
 
+            elif move_line.move_id.move_type in ('out_refund', 'in_refund'):
+                map_move_sale_line[move_line.id] = self.env['sale.order.line'].search([('order_id', '=', sale_order.id),
+                                                                                       ('price_unit', '=', price),
+                                                                                       ('product_id', '=', move_line.product_id.id),
+                                                                                       ('is_expense', '=', True),
+                                                                                       ], limit=1)
```

Comment:
![image](https://user-images.githubusercontent.com/44770049/234605166-890e2adc-f6a7-4730-b38e-d2802c5043f4.png) :vomiting_face: 
```suggestion
                map_move_sale_line[move_line.id] = self.env['sale.order.line'].search([
                    ('order_id', '=', sale_order.id),
                    ('price_unit', '=', price),
                    ('product_id', '=', move_line.product_id.id),
                    ('is_expense', '=', True),
                ], limit=1)
```